File: SDK/java/src/main/java/com/reloop/verifio/errors/InsufficientCreditsException.java
package com.reloop.verifio.errors;

public class InsufficientCreditsException extends VerifioException {
    private final Integer remaining;
    private final Integer required;

    public InsufficientCreditsException(String message, Integer remaining, Integer required) {
        super(message != null ? message : "Insufficient credits", 402, "INSUFFICIENT_CREDITS");
        this.remaining = remaining;
        this.required = required;
    }

    public Integer getRemaining() {
        return remaining;
    }

    public Integer getRequired() {
        return required;
    }
}

File: SDK/php/src/Exceptions/InsufficientCreditsException.php
<?php

namespace Verifio\Exceptions;

class InsufficientCreditsException extends VerifioException
{
    protected ?int $remaining;
    protected ?int $required;

    public function __construct(?string $message = null, ?int $remaining = null, ?int $required = null)
    {
        parent::__construct($message ?? "Insufficient credits", 402, "INSUFFICIENT_CREDITS");
        $this->remaining = $remaining;
        $this->required = $required;
    }

    public function getRemaining(): ?int
    {
        return $this->remaining;
    }

    public function getRequired(): ?int
    {
        return $this->required;
    }
}
